src/test/scala/chiselTests/simulator/scalatest/WithTestingDirectorySpec.scala

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests.simulator.scalatest
+
+import chisel3._
+import chisel3.simulator.scalatest.WithTestingDirectory
+import chisel3.simulator.DefaultSimulator._
+import java.nio.file.FileSystems
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+import scala.reflect.io.Directory
+
+class WithTestingDirectorySpec extends AnyFunSpec with Matchers with WithTestingDirectory {
+
+  class Foo extends Module {
+    stop()
+  }
+
+  describe("A test suite mixing in WithTestingDirectory") {
+    it("should generate a directory structure derived from the suite and test name") {
+
+      val directory = Directory(
+        FileSystems
+          .getDefault()
+          .getPath(
+            "test_run_dir",
+            "WithTestingDirectorySpec",
+            "A_test_suite_mixing_in_WithTestingDirectory",
+            "should_generate_a_directory_structure_derived_from_the_suite_and_test_name"
+          )
+          .toFile()
+      )
+      directory.deleteRecursively()
+
+      simulate(new Foo()) { _ => }
+
+      info(s"found expected directory: '$directory'")
+      assert(directory.exists)
+      assert(directory.isDirectory)
+
+      val allFiles = directory.deepFiles.toSeq.map(_.toString).toSet
+      for (
+        file <- Seq(
+          directory.toFile.toString + "/workdir-default/Makefile",
+          directory.toFile.toString + "/primary-sources/Foo.sv"
+        )
+      ) {
+        info(s"found expected file: '$file'")
+        allFiles should contain(file)
+      }
+    }
+    it("should generate another directory, too") {
```

Comment:
What if the test uses a non-ASCII character like a Chinese character or an emoji?

Do we want to `java.text.Normalizer` it or just leave it be?

Reply:
Maybe. Let me play with this. I'm also kind of of the opinion that maybe we don't mangle this at all. Emojis or CJK characters all seem reasonable to _not_ mangle as I think most filesystems will support those?

Reply:
I like the ` ` to `_` mangling and I agree I'm not sure we should do any other mangling, just wanted to float the idea. It's also reasonable for users to avoid such characters if they do cause problems but obviously spaces are to be expected.

Reply:
Switched to mangle all whitespace to `-` in: f39550b8b This includes tests that both emoji and CJK characters (technically just Chinese (or Japanese Kanji)) work.

Reply:
I don't object by why `-` over `_`?